includes/reports/overview/services/class-sensei-reports-overview-service-courses.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * File containing the Sensei_Reports_Overview_Service_Courses class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Courses overview service class.
+ *
+ * @since 4.5.0
+ */
+class Sensei_Reports_Overview_Service_Courses {
```

Comment:
As the service is pretty specific, we can name it more specifically :)
Maybe something like `Sensei_Reports_Overview_Service_Total_Average_Progress`?
And maybe rename `get_total_average_progress` to just `get` to remove duplication while calling it :)

Reply:
@merkushin  
1. I've moved the tests to the service https://github.com/Automattic/sensei/pull/5077/commits/c80d67ee0f6f3a44c08c2ba73e41fcb1c7872660

2. Hmh since we have data-provider and the main list-table I taught it is better to have something more generic where we can keep related methods, I actually like it this way. How strong are you about this?